src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/EnumerableExtensions.cs

```diff
@@ -99,7 +99,10 @@ IEnumerable<TResult> _()
 
         public static async Task ForEachAsync<T>(this IEnumerable<T> list, Func<T, Task> func)
         {
-            await Task.WhenAll(list.Select(func));
+            foreach (var item in list)
+            {
+                await func(item);
```

Comment:
Ah, was this causing threading issues?

Reply:
Yes I started getting this when the task is a database operation

`
System.InvalidOperationException: A second operation started on this context before a previous operation completed. This is usually caused by different threads using the same instance of DbContext. For more information on how to avoid threading issues with DbContext, see https://go.microsoft.com/fwlink/?linkid=2097913.
   at Microsoft.EntityFrameworkCore.Internal.ConcurrencyDetector.EnterCriticalSection()
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.AsyncEnumerator.MoveNextAsync()
   at Microsoft.EntityFrameworkCore.Query.ShapedQueryCompilingExpressionVisitor.SingleOrDefaultAsync[TSource](IAsyncEnumerable`1 asyncEnumerable, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.ShapedQueryCompilingExpressionVisitor.SingleOrDefaultAsync[TSource](IAsyncEnumerable`1 asyncEnumerable, CancellationToken cancellationToken)
   at GovUk.Education.ExploreEducationStatistics.Admin.Services.ReplacementService.ReplaceFootnoteSubject(Guid footnoteId, Guid originalSubjectId, Guid replacementSubjectId) in C:\Users\ben\dev\explore-education-statistics\src\GovUk.Education.ExploreEducationStatistics.Admin\Services\ReplacementService.cs:line 552
   at GovUk.Education.ExploreEducationStatistics.Admin.Services.ReplacementService.ReplaceLinksForFootnote(FootnoteReplacementPlanViewModel replacementPlan, Guid originalSubjectId, Guid replacementSubjectId) in C:\Users\ben\dev\explore-education-statistics\src\GovUk.Education.ExploreEducationStatistics.Admin\Services\ReplacementService.cs:line 535
   at GovUk.Education.ExploreEducationStatistics.Common.Extensions.EnumerableExtensions.ForEachAsync[T](IEnumerable`1 list, Func`2 func)
`

Reply:
Cool, fair enough!